### PR TITLE
Add sphericaltexture_plugin to default object feature plugins

### PR DESCRIPTION
**Pull request by k-dominik**
CC: @oanegros

this is ready now. There were some complications on windows which I didn't follow to the root cause, but, some versions of `pyshtools` (a dependency of `sphericaltextures`) would clash with `pytorch`. It looks like these issues have been resolved with pytorch 2.4.*.

Also removed `inferno` (resulted in conflicts, but since it's not used in our code it was a good reminder to actually remove it).

The Object Classification tests now also calculate the spherical texture features, so the reference data needed regenerating.

- [ ] Format code and imports.
- [ ] Add docstrings and comments.
- [ ] Add tests.
- [ ] Update [user documentation](https://github.com/ilastik/ilastik.github.io).
- [ ] Reference relevant issues and other pull requests.
- [ ] Rebase commits into a logical sequence.

### CODE REVIEW

File: conda-recipe/meta.yaml [hunk 1]
```diff
@@ -67,9 +67,12 @@ outputs:
         - z5py
         - zarr 2.*
       run_constrained:
-        - mkl !=2024.1.0
+        - mkl <2024.1.0
```
Comment (btbest) [inline conda-recipe/meta.yaml:70]: :(

File: tests/test_ilastik/test_workflows/testObjectClassificationGui.py [hunk 2]
```diff
@@ -253,6 +252,13 @@ def impl():
                 for plugin in features
             }
 
+            # in order to also calculate neighborhood features, we need to set one
+            # margin to non-zero
+            for plugin in features:
+                for _, feature_descr in features[plugin].items():
+                    if "margin" in feature_descr:
+                        feature_descr["margin"] = (3, 3, 3)
```
Comment (btbest) [inline tests/test_ilastik/test_workflows/testObjectClassificationGui.py:260]: I'm not familiar with what the properties of the feature description mean, but why do we need to overwrite a value if it exists?

**Review by btbest — APPROVED** [review]

File: conda-recipe/meta.yaml (hunk 1 above)
Comment (k-dominik) [reply conda-recipe/meta.yaml:70]: yeah, hope this is resolved soon (related: https://github.com/pytorch/pytorch/issues/129310)

File: tests/test_ilastik/test_workflows/testObjectClassificationGui.py (hunk 2 above)
Comment (k-dominik) [reply tests/test_ilastik/test_workflows/testObjectClassificationGui.py:260]: yes, this setting is to be overridden by the UI (hence, manually in test). The existence of the "margin" key indicates that a feature is computed on the object + some neighbourhood around it. The logic will hopefully soon change, but for now it's like that - and for the tests to trigger the computation of those neighbourhood features, there needs to be a non-zero value (all "neighbourhood features" have a 0 there as default).

**Pull request merged**